### ISSUE TITLE
Large vessels & hangar

### ISSUE THREAD

**Issue by ErikDeBruijn** [daid/EmptyEpsilon#33]
Each larger vessel (class > 6) with docking abilities should have a hangar deck. 

I'd like to make it so that only when an engineering crewmember is in the hangar, present ships will be repaired.

**Comment by daid** [comment]
class > 6? everything is class 10 by default. Except for the huge battlestation, which is 11. And note that this docking mechnism needs some refinement.

**Comment by nallath** [comment]
We will need to refactor the power stuff anyway; All ships have 1000 power I believe, which is weird. Fighters should drain power from the larger ships, but it would be weird if a fighter uses as much power as a big honking space cruiser.

**Comment by ErikDeBruijn** [comment]
@nallath good point wrt energy drain of main ship. I'd love to add a SYS_HangarDeck. Actually I already did... but I had some segfaults so didn't want to commit code that didn't even work well on my own systems. Now I can finally test on both Mac and Linux and make code that works across at least two platforms.

**Comment by Jhovall** [comment]
Although this is quite an old issue, are there any updates on it? And how to assign classes to the ships? IIRC we can't assign a class in the shiptemplates.
Will there be a docking possibility on a specific place? just like the engines and beams are placed?

**Comment by daid** [comment]
Currently, ships can dock other ships, if the ship you are trying to dock to has a higher "size class", and the ship is not your enemy.

The default size class (when it is not set) is 10.
Currently only the "Battlestation" has a size class specified. So other ships can dock to a "Battlestation"

```
template = ShipTemplate():setName("Battlestation"):setModel("Ender Battlecruiser")
template:setSizeClass(11)
```

Note that being docked means that your energy will recharge, no matter what you are docked too. Other then that, it should work fine, and your docking target can still fly, or even use it's jumpdrive.

**Comment by Jhovall** [comment]
That is really nice to know. (And usefull for our larps), I will implement this in my new ship templates.
Does the class do anything other than being able to dock if class > 6?

Questions which I could find out myself, but not able to at the moment:
And is there a specific docking point (I guess not) and does the docking-ship fly along with the main ship? Does the docked ship dissappear?

**Comment by Jhovall** [comment]
Also,

Might it be possible that a ship is too ' large'  to dock to another ship, but still too small to let another ship dock.

a.k.a. Some of my ships do have a hangar deck for shuttles, others don't, but are still large.

**Comment by daid** [comment]
Currently that cannot be done yet. I'll have to change the "docking rules" for that.

There are no docking points. Docking works by "bumping" into each other, where ever the ships meet is where the docking happens.

**Comment by daid** [comment]
The sizeClass is only used for docking right now.

**Comment by Jhovall** [comment]
Well, my feature request is that there is a possibility for docking points at ships and spacestations.
And a docking rule for ships/spacestations, so you can say who can dock where.

I have a large spacestation with several hangars and I only want to allow the ships dock at one of these hangars. (It would be ugly if they docked somewhere else).
